public/js/src/module/store.js

```diff
@@ -296,47 +306,49 @@ const surveyStore = {
                         .filter( existing => binaryDefaultKeys.indexOf( existing ) < 0 );
                 }
 
-                _transformExternalDataToXmlStr( survey );
+                let update = Object.assign( {}, _serializeSurvey( survey ) );
+
+                // Note: if survey.resources = undefined/null, do not store empty array
+                // as it means there are no resources to store (and load) - they may be added later by form-cache.js
+                if ( survey.resources ) {
+                    update.resources = resourceKeys;
+                }
+
+                if ( survey.binaryDefaults ) {
+                    update.resources = binaryDefaultKeys;
+                }
 
                 // Update the existing survey
-                return server.surveys.update( {
-                    form: survey.form,
-                    model: survey.model,
-                    enketoId: survey.enketoId,
-                    hash: survey.hash,
-                    theme: survey.theme,
-                    // Note: if survey.resources = undefined/null, do not store empty array
-                    // as it means there are no resources to store (and load) - they may be added later by form-cache.js
-                    resources: survey.resources ? resourceKeys : undefined,
-                    binaryDefaults: survey.binaryDefaults ? binaryDefaultKeys : undefined,
-                    maxSize: survey.maxSize,
-                    externalData: survey.externalData,
-                    branding: survey.branding
-                } );
+                return server.surveys.update( update );
```

Comment:
This set of changes was primarily about minimizing the difference and potential drift between create/update logic, as it was quite confusing in testing why `lastSavedRecord` was being set once then dropped thereafter.